main.py

```diff
@@ -700,6 +710,28 @@ def delete_schedule(self, request: Request) -> JSONResponse:
         log.debug("delete_schedule result %s %s", result, status)
         return JSONResponse(result, status_code=status)
 
+    def _check_no_tag_duplication(self, evc_dict: dict):
```

Comment:
Can you refactor to pass the arguments `evc_id: str, uni_a: Optional[UNI] = None, uni_z: Optional[UNI]` you need instead of a dict? It'd make the intention clear and slightly easier to test and understand this method, and in the future less error prone if someone has to also update or refactor it.

Reply:
Got it, changed.